### PR TITLE
Changed TrustChain so it's a single chain instead of multiple fragmented communities

### PR DESCRIPTION
**Pull request by devos50**
In this PR, I refactored the communities that were subclassing the TrustChain overlay. Instead, we have a single ledger now and we create separate overlays and communicate with the TrustChain ledger instead. See the IPv8 pull request for more information.

I also updated the master peers of the market + tunnel community. I promise this will be the last time :). Finally, I fixed a bug during the tests where IPv8 would talk with the real world.

Requires https://github.com/Tribler/py-ipv8/pull/96 to be merged first.

I've written a few experiments to verify that everything works as expected with this PR:
- TrustChain: https://jenkins.tribler.org/job/Experiment_trustchain_integration/130/
- Market: https://jenkins.tribler.org/job/market_experiment_validation/7/
- Anonymous downloading: https://jenkins.tribler.org/job/pers/job/anon_downloads_devos50/128/

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -9,4 +9,4 @@
 	url = https://github.com/spesmilo/electrum
 [submodule "py-ipv8"]
 	path = Tribler/pyipv8
-	url = https://github.com/Tribler/py-ipv8.git
+	url = https://github.com/devos50/py-ipv8.git
```
Comment (qstokkink) [inline .gitmodules]: ❗️

Comment (devos50) [reply .gitmodules]: I know and this will be removed after the review and when the IPv8 PR is merged.

**Comment by qstokkink** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by ichorid** [comment]
Style is good and I love how the net change of lines of code is negative ))

**Comment by synctext** [comment]
Big conceptual step forward.

**Comment by devos50** [comment]
There seems to be an error during the payouts when running this PR in my private testnet. I will change the status of this PR to WIP again.

**Comment by devos50** [comment]
I managed to fix the error with the payouts but this PR is not reliable enough yet. I will spend some time writing application tests and making sure that this does not break anything.

**Comment by devos50** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

File: Tribler/Core/APIImplementation/LaunchManyCore.py [hunk 2]
```diff
@@ -40,13 +34,19 @@
                                      STATE_START_TORRENT_CHECKER, STATE_START_REMOTE_TORRENT_HANDLER,
                                      STATE_START_API_ENDPOINTS, STATE_START_WATCH_FOLDER, STATE_START_CREDIT_MINING)
 from Tribler.dispersy.util import blockingCallFromThread, blocking_call_on_reactor_thread
+from Tribler.pyipv8.ipv8.attestation.trustchain.community import TrustChainCommunity
```
Comment (qstokkink) [inline Tribler/Core/APIImplementation/LaunchManyCore.py]: This should be imported when loading the community.

File: Tribler/Core/Modules/wallet/tc_wallet.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+            self._logger.info("Checking for block with id %s and num %d", pub_key.encode('hex'), sequence_number)
+            db_block = self.trustchain.persistence.get(pub_key, sequence_number)
+            if db_block:
+                monitor_lc.stop()
+                monitor_deferred.callback(db_block)
+
+        if block:
+            return succeed(block)
+
+        monitor_lc = LoopingCall(check_has_block)
```
Comment (qstokkink) [inline Tribler/Core/Modules/wallet/tc_wallet.py]: This is not a TaskManager managed task and can leak.

File: Tribler/community/triblertunnel/community.py [hunk 4]
```diff
@@ -271,7 +273,7 @@ def update_torrent(self, peers, handle, download):
     def get_peer_from_address(self, address):
         circuit_peer = None
         for peer in self.get_peers():
-            if peer.address == address:
+            if peer.address == address and peer.public_key.key_to_bin().startswith("LibNaCLPK:"):
```
Comment (qstokkink) [inline Tribler/community/triblertunnel/community.py]: You can remove this now, fixed in IPv8.

File: Tribler/community/market/community.py [hunk 5]
```diff
@@ -1657,49 +1638,43 @@ def send_transaction_completed(self, transaction, block):
...
-        linked_block = self.market_database.get_linked(block) or block
-        self.send_block_pair(block, linked_block, address=self.lookup_ip(match_payload.matchmaker_trader_id))
+        linked_block = self.trustchain.persistence.get_linked(block) or block
 
-    def on_transaction_completed_message(self, block1, block2):
-        tx_dict = block1.transaction
+        global_time = self.claim_global_time()
+        dist = GlobalTimeDistributionPayload(global_time).to_pack_list()
+        payload = HalfBlockPairPayload.from_half_blocks(block, linked_block).to_pack_list()
+        packet = self._ez_pack(self._prefix, 22, [dist, payload], False)
```
Comment (ichorid) [inline Tribler/community/market/community.py]: 22 = Magic number ?

Comment (devos50) [reply Tribler/community/market/community.py]: Yeah, this is a convention that I followed from the Dispersy era. I will use some defined constants in a subsequent PR 👍

**Pull request merged**